uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/controllers/calendar-ctrl.js

```diff
@@ -0,0 +1,283 @@
+define([
+  'leave-absences/shared/modules/controllers',
+  'common/lodash',
+  'common/moment',
+  'leave-absences/shared/models/absence-period-model',
+  'leave-absences/shared/models/absence-type-model',
+  'leave-absences/shared/models/public-holiday-model',
+], function (controllers, _, moment) {
+  'use strict';
+
+  controllers.controller('CalendarCtrl', ['$timeout', 'shared-settings', 'AbsencePeriod', 'AbsenceType',
+    'PublicHoliday', 'OptionGroup', controller]);
+
+  function controller($timeout, sharedSettings, AbsencePeriod, AbsenceType, PublicHoliday, OptionGroup) {
```

Comment:
Yes we are doing it in few other places too like `myLeaveReport`, `managerLeaveRequests`.
To be honest I feel separating the function makes the code more clean.